mteb/abstasks/AbsTaskReranking.py

```diff
@@ -1,98 +1,216 @@
+            Semantically, it should contain dict[split_name, dict[sample_id, str]] or dict[split_name, dict[sample_id, list[str]]] for conversations
+            E.g. {"test": {"q1": "query"}}
+            or {"test": {"q1": ["turn1", "turn2", "turn3"]}}
+
+        self.relevant_docs: dict[str, dict[str, dict[str, int]]]
+            Semantically, it should contain dict[split_name, dict[sample_id, dict[doc_id, score]]]
+            E.g.: {"test": {"q1": {"document_one": 1}}}
+
+        self.top_ranked: dict[str, dict[str, list[str]]] or dict[str, dict[str, dict[str, float]]]
+            Semantically, it should contain dict[split_name, dict[sample_id, list[doc_id]]] or dict[split_name, dict[sample_id, dict[doc_id, score]]]
+            E.g.: {"test": {"q1": ["document_one", "document_two"]}} or {"test": {"q1": {"document_one": 1, "document_two": 0.5}}}
+
+    Old Format:
+    -----------
     self.load_data() must generate a huggingface dataset with a split matching self.metadata_dict["eval_splits"], and assign it to self.dataset. It must contain the following columns:
         query: str
         positive: list[str]
         negative: list[str]
     """
 
     def __init__(self, **kwargs):
-        super().__init__(**kwargs)
+        super(AbsTaskRetrieval, self).__init__(**kwargs)
+
+    def load_data(self, **kwargs):
+        if self.data_loaded:
+            return
+
+        if self.metadata.name in OLD_FORMAT_RERANKING_TASKS:
+            self.dataset = datasets.load_dataset(**self.metadata_dict["dataset"])  # type: ignore
+            self.dataset_transform()
+        else:
+            # use AbsTaskRetrieval default to load the data
+            # TODO: need to make sure top_ranked comes back
+            return super().load_data(**kwargs)
+
+    def process_example(self, example: dict, split: str, query_idx: int) -> dict:
+        """Process a single example from the dataset."""
+        query = example["query"]
+        positive_docs = example["positive"]
+        negative_docs = example["negative"]
+
+        query_id = f"{split}_query{query_idx}"
+
+        # Initialize the structures for this example
+        example_data = {
+            "query_id": query_id,
+            "query": query,
+            "doc_ids": [],
+            "doc_texts": [],
+            "relevance_scores": [],
+        }
+
+        for i, pos_doc in enumerate(sorted(positive_docs)):
+            doc_id = f"{query_id}_positive_{i}"
+            example_data["doc_ids"].append(doc_id)
+            example_data["doc_texts"].append(pos_doc)
+            example_data["relevance_scores"].append(1)
+
+        for i, neg_doc in enumerate(sorted(negative_docs)):
+            doc_id = f"{query_id}_negative_{i}"
+            example_data["doc_ids"].append(doc_id)
+            example_data["doc_texts"].append(neg_doc)
+            example_data["relevance_scores"].append(0)
+
+        return example_data
+
+    def dataset_transform(self):
```

Comment:
wouldn't this be overwritten by the tasks own dataset transform?

Reply:
Ah shoot, good point. Okay, will need to rename this function, thanks!